src/main/java/org/truffleruby/core/exception/CoreExceptions.java

```diff
@@ -1031,13 +1031,17 @@ public RubyException ioError(IOException exception, Node currentNode) {
 
     // RangeError
 
-    @TruffleBoundary
     public RubyException rangeError(long code, RubyEncoding encoding, Node currentNode) {
         return rangeError(
-                StringUtils.format("invalid codepoint %x in %s", code, encoding.encoding),
+                rangeErrorMessage(code, encoding.encoding),
                 currentNode);
     }
 
+    @TruffleBoundary
+    private String rangeErrorMessage(long code, Encoding encoding) {
+        return StringUtils.format("invalid codepoint %x in %s", code, encoding);
```

Comment:
`StringUtils.format` is already a boundary so there is no need for this method. I'll fix it.
BTW, the rationale to have the boundary early is to avoid having two boundaries instead like with this change (format & encodeRope).

Performance-wise, the critical thing is to not need a backtrace and that PE sees the Throwable: `throw new RaiseException`.
The RubyException allocation does not matter much.